File: src/commands/store_hours.js
// commands/store_hours.js — Store opening hours
import { rpc } from "@/lib/apiClient";

export const getStoreHours = (storeId) =>
  rpc("get_store_hours", { store_id: storeId });

export const upsertStoreHours = (storeId, hours) =>
  rpc("upsert_store_hours", { store_id: storeId, hours });
// hours: Array of { store_id, day_of_week (0-6), is_open, open_time?, close_time? }
// day_of_week: 0=Sun, 1=Mon … 6=Sat

File: src-tauri/src/models/store_hours.rs
// ============================================================================
// STORE HOURS MODELS
// ============================================================================

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Clone, sqlx::FromRow)]
pub struct StoreHour {
    pub id:          i32,
    pub store_id:    i32,
    pub day_of_week: i16,     // 0=Sun … 6=Sat
    pub is_open:     bool,
    pub open_time:   Option<String>,  // "HH:MM"
    pub close_time:  Option<String>,  // "HH:MM"
}

#[derive(Debug, Deserialize)]
pub struct UpsertStoreHourDto {
    pub store_id:    i32,
    pub day_of_week: i16,
    pub is_open:     bool,
    pub open_time:   Option<String>,
    pub close_time:  Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct BulkUpsertStoreHoursDto {
    pub store_id: i32,
    pub hours:    Vec<UpsertStoreHourDto>,
}
